pkg/handlers/ghcapi/move_task_order.go

```
@@ -52,12 +54,22 @@ type UpdateMoveTaskOrderStatusHandlerFunc struct {
 func (h UpdateMoveTaskOrderStatusHandlerFunc) Handle(params movetaskorderops.UpdateMoveTaskOrderStatusParams) middleware.Responder {
 	session, logger := h.SessionAndLoggerFromRequest(params.HTTPRequest)
 	eTag := params.IfMatch
+	var serviceItemCodes []models.ReServiceCode
+
+	for _, serviceItemCode := range params.ServiceItemCodes {
+		if string(serviceItemCode) == string(models.ReServiceCodeCS) {
+			serviceItemCodes = append(serviceItemCodes, models.ReServiceCodeCS)
+		}
+		if string(serviceItemCode) == string(models.ReServiceCodeMS) {
+			serviceItemCodes = append(serviceItemCodes, models.ReServiceCodeMS)
+		}
+	}
```

I know in the form we require at least one service item, should we throw an error if there are no service item code params?

I think you can simplify these comparisons a little bit like:
```go
for _, serviceItemCode := range params.ServiceItemCodes {
    switch models.ReServiceCode(serviceItemCode) {
        case models.ReServiceCodeCS:
            serviceItemCodes = append(serviceItemCodes, models.ReServiceCodeCS)
        case models.ReServiceCodeMS:
            serviceItemCodes = append(serviceItemCodes, models.ReServiceCodeMS)
    }
}
```

---

Hm, yeah I suppose throwing an error in that case makes sense since it would make the business rules consistent between the interface and the end point. I'll add that in.

---

Done. Went ahead and handled the error condition by making the parameter required in swagger.